support-workers/src/main/scala/com/gu/emailservices/SupporterPlusEmailFields.scala

```diff
@@ -0,0 +1,81 @@
+package com.gu.emailservices
+
+import com.gu.emailservices.SubscriptionEmailFieldHelpers.{formatDate, hyphenate, mask}
+import com.gu.salesforce.Salesforce.SfContactId
+import com.gu.support.workers._
+import com.gu.support.workers.states.SendThankYouEmailState.{
+  SendThankYouEmailContributionState,
+  SendThankYouEmailSupporterPlusState,
+}
+import org.joda.time.DateTime
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SupporterPlusEmailFields(
+    getMandate: String => Future[Option[String]],
+    created: DateTime,
+) {
+
+  def build(
+      state: SendThankYouEmailSupporterPlusState,
+  )(implicit ec: ExecutionContext): Future[EmailFields] = {
+    getPaymentFields(
+      state.paymentMethod,
+      state.accountNumber,
+      created,
+    ).map { paymentFields =>
+      val fields = List(
+        "EmailAddress" -> state.user.primaryEmailAddress,
+        "created" -> created.toString,
+        "amount" -> state.product.amount.toString,
+        "currency" -> state.product.currency.iso,
+        "edition" -> state.user.billingAddress.country.alpha2,
+        "name" -> state.user.firstName,
+        "product" -> s"${state.product.billingPeriod.toString.toLowerCase}-supporter-plus",
+      ) ++ paymentFields
```

Comment:
do we need to add a field to say whether they should get the "you have benefits" or "you don't have benefits" email?

Reply:
As above, anyone on this product will receive benefits